tests/test_custom_objects_rest.py

```diff
@@ -0,0 +1,100 @@
+    """Test that all fields can be replicated for a stream that is a custom object"""
+
+    # Note: custom record as seen Aug 9, 2023
+    # TODO do we need to hard code start date to 08-04?
+    # 'CreatedDate': '2023-08-04T20:13:50.000000Z'
+    # 'SystemModstamp': '2023-08-04T20:13:50.000000Z',
+    # 'LastModifiedDate': '2023-08-04T20:13:50.000000Z',
+    # 'LastReferencedDate': '2023-08-09T20:19:45.000000Z',
+    # 'LastViewedDate': '2023-08-09T20:19:45.000000Z',
+    start_date =  (datetime.now() + timedelta(days=-1)).strftime("%Y-%m-%dT00:00:00Z")
+
+    @staticmethod
+    def expected_sync_streams():
+        return {
+            'TapTester__c',  # 1 created record present, spike on CRUD?
+            # 'TapTester__Share',  # TODO no data, fix or omit?
+        }
+
+    def custom_objects_test(self):
+        """
+        Verify that all fields can be replicated for a custom  stream
+
+        PREREQUISITE
+        Create a custom object and at least one data record for that cusotm object
+        """
+
+        expected_streams = self.expected_sync_streams()
+
+        # instantiate connection
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                                          if catalog.get('stream_name') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, test_catalogs, select_all_fields=True,
+        )
+
+        # grab metadata after performing table-and-field selection to set expectations
+        #   used for asserting all fields are replicated
+        stream_to_all_catalog_fields = dict()
+        for catalog in test_catalogs:
+            stream_id, stream_name = catalog['stream_id'], catalog['stream_name']
+            catalog_entry = menagerie.get_annotated_schema(conn_id, stream_id)
+            fields_from_field_level_md = [md_entry['breadcrumb'][1]
+                                          for md_entry in catalog_entry['metadata']
+                                          if md_entry['breadcrumb'] != []]
+            stream_to_all_catalog_fields[stream_name] = set(fields_from_field_level_md)
+
+        # run initial sync
+        record_count_by_stream = self.run_and_verify_sync(conn_id)
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_keys = stream_to_all_catalog_fields.get(stream)
+
+                # collect actual values
+                data = synced_records.get(stream)
+                record_messages_keys = [set(row['data'].keys()) for row in data['messages']
+                                        if row['action'] == 'upsert']
```

Comment:
hmmm. I'm not sure I understand this.  Looks like you are going to end up with a list of multiple sets of keys.  I would think we want a single list of all the keys in the messages.  Maybe you are planning doing a union of the members of the list next?

Reply:
Related to below comment.